security-analysis/security-analysis-api/src/main/java/com/powsybl/security/tools/SecurityAnalysisTool.java

```diff
@@ -217,9 +217,11 @@ static SecurityAnalysisInput buildPreprocessedInput(SecurityAnalysisExecutionInp
     }
 
     private static SecurityAnalysisExecutionBuilder createBuilder(PlatformConfig platformConfig) {
+        String providerName = platformConfig.getOptionalModuleConfig(MODULE_CONFIG_NAME_PROPERTY)
+                .flatMap(c -> c.getOptionalStringProperty(DEFAULT_SERVICE_IMPL_NAME_PROPERTY))
+                .orElse(null);
         return new SecurityAnalysisExecutionBuilder(() -> ExternalSecurityAnalysisConfig.load(platformConfig),
-            platformConfig.getModuleConfig(MODULE_CONFIG_NAME_PROPERTY).getStringProperty(DEFAULT_SERVICE_IMPL_NAME_PROPERTY),
```

Comment:
I changed the behaviour here, as it was throwing an exception before if the provider name was not specified.